ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/CastStringToDate.java

```diff
@@ -116,14 +118,16 @@ public void evaluate(VectorizedRowBatch batch) {
 
   private void evaluate(LongColumnVector outV, BytesColumnVector inV, int i) {
     String dateString = new String(inV.vector[i], inV.start[i], inV.length[i], StandardCharsets.UTF_8);
-    if (dateParser.parseDate(dateString, sqlDate)) {
+    try {
+      Date utilDate = Date.valueOf(dateString);
```

Comment:
Is this vectorized expression generated for `GenericUDFDate`? It seems that UDF still relies on DateParser in this branch: https://github.com/apache/hive/blob/b7f3e8ef399f510c3a6780209ebc688ef0acee8f/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFDate.java#L112

Since the change may have an effect on parsing of other dates, it may be good to have a test run. It seems we have a jenkins active for branch-2 patches: http://130.211.9.232/job/hive-precommit/job/branch-2/ . I am wondering whether a PR against branch-2 will trigger it or whether we can trigger it manually. Could you verify?

Reply:
Yes this vectorized expression is generated for `GenericUDFDate`.  I don't see any way to trigger precommit job manually for branch-2. @kgyrtkirk  Do you know how can I trigger ptest for branch-2?